iroh-sync/src/net/codec.rs

```diff
@@ -260,10 +293,16 @@ mod tests {
         let (mut bob_reader, mut bob_writer) = tokio::io::split(bob);
         let bob_replica_store_task = bob_replica_store.clone();
         let bob_task = tokio::task::spawn(async move {
-            run_bob::<store::memory::Store, _, _>(
+            run_bob::<store::memory::Store, _, _, _, _>(
                 &mut bob_writer,
                 &mut bob_reader,
-                bob_replica_store_task,
+                |namespace, _| {
+                    futures::future::ready(
+                        bob_replica_store_task
+                            .open_replica(&namespace)
```

Comment:
this is a blocking call, we should probably mark it as such

Reply:
deferred for later: https://github.com/n0-computer/iroh/issues/1502